File: InfrastructureVB/Data/ICRUD.vb
Namespace Data
    Public Interface ICRUD(Of TEntity, TKey)
        Inherits IDisposable

        Function [Get](id As TKey) As TEntity

        Function Create(entity As TEntity) As TKey

        Sub Update(entity As TEntity)

        Sub Delete(id As TKey)
    End Interface
End Namespace

File: Infrastructure/Data/ICRUD.cs
﻿using System;

namespace Infrastructure.Data
{
    public interface ICRUD<TEntity, TKey> : IDisposable
    {
        TEntity Get(TKey id);
        TKey Create(TEntity entity);
        void Update(TEntity entity);
        void Delete(TKey id);
    }
}
